### DISCUSSION TITLE
port /icubsrv is not responding   &&    port /icub-head is not responding

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#620] [Q&A] [answered]
Dear all, when i compile the app icubSim below on the docker, it works fine, 
https://github.com/vvv-school/tutorial_joint-interface

optJoint.put("remote","/icubSim/left_arm");
--->
optJoint.put("remote","/icub/left_arm");

but when i change icubSim to icub as above, compile and want to run yarpmanager again, i got the following error message
(i install the hardware correctly except that the cpu and motor lighting is red)

12:43 # yarpmanager
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
[ERROR] |yarp.os.Property| cannot read from ymanager.ini
[INFO] |yarp.os.Port|/tmp/port/1| Port /tmp/port/1 active at tcp://172.17.0.2:10002/
[WARNING] Cannot connect to /icubsrv
[ERROR] ClusterWidget: port /icubsrv is not responding
[INFO] |yarp.os.Port|/tmp/port/1| Port /tmp/port/1 active at tcp://172.17.0.2:10002/
[WARNING] Cannot connect to /icub-head
[ERROR] ClusterWidget: port /icub-head is not responding
QObject::connect: No such slot ApplicationViewWidget::onBuilderFloatChanged(bool)
QObject::connect:  (receiver name: 'ApplicationViewWidget')
QRectF(-71,-16 142x32)

**Comment by pattacini** [comment]
Hi @jimingda 

If you aim to run something on the robot, you ought to ensure that the following requirements are met:
1. Warm up the network composed of the robot and the laptop: both machines ON (on the robot both the CPU and the motors need to be ON) and with a good ETH connection established so that you can do `ping icubsrv` and `ping icub-head` from one to another.
2. `yarpserver` needs to run on `icubsrv`. You can launch it from the console or from within the `yarpmanager`. Once done, you can check that it is visible by yielding `yarp where` on both machines in the console.
3. Two options:
    1. You can now _ssh_ into `icub-head`, build the app `tutorial_joint-interface` therein without relying on `yarpamanager`. Then, do:
        1. Launch `yarprobotiterface` from the terminal and wait until the robot is ready.
        2. Launch the app from a second terminal.
    3. In case you want to launch the app remotely relying on `yarpmanager`, do:
        1. On `icubsrv`, from within `yarpmanager`, launch the `yarprun` instances (I assume you're familiar with `yarpmanager`)
        2. Launch `yarprobotinterface` from the GUI and wait.
        3. Launch the app from the GUI.

**Reply by pattacini** [reply]
You may find here some useful recaps: https://icub-tech-iit.github.io/documentation/icub_starter_kits/first_steps/#first-start-with-the-icub.

**Reply by jimingda** [reply]
Thanks ugo,  I tryed two ways, both under the circumstances that  the real iCub ETH connected and the cable is powered on 40V & 10A.

1.
I launch yarpserver on one terminal seperately, and then launch yarp manager on another terminal, get the following messages, it seems like something wrong with /icub-head connection

icub@icubsrv:~$ yarp where
Looking for name server on 172.17.0.1, port number 10000
If there is a long delay, try:
  yarp conf --clean
Name server /icub is available at ip 172.17.0.1 port 10000
icub@icubsrv:~$ yarpmanager
[ERROR] |yarp.os.Property| cannot read from ymanager.ini
[INFO] |yarp.os.Port|/tmp/port/1| Port /tmp/port/1 active at tcp://192.168.31.30:10002/
[WARNING] Cannot connect to /icubsrv
[ERROR] ClusterWidget: port /icubsrv is not responding
[INFO] |yarp.os.Port|/tmp/port/1| Port /tmp/port/1 active at tcp://192.168.31.30:10002/
[WARNING] Cannot connect to /icub-head
[ERROR] ClusterWidget: port /icub-head is not responding
[INFO] |yarp.os.Port|/icubsrv| Port /icubsrv active at tcp://192.168.31.30:10003/
[INFO] Yarprun successfully started on port:  /icubsrv

2.
https://icub-tech-iit.github.io/documentation/icub_starter_kits/first_steps/#first-start-with-the-icub
I followed the steps launch yarpserver within the yarpmanager directly, and i successfully launch the yarpserver, but when run selected nodes, 

On nodes :
    please select both icubsrv and icub-head
    click GREEN play button. 
    Both TV icon should turn from RED to GREEN

only icubsrv icon turn from RED to GREEN and the footer Messages on yarpmanager shows:
[MSG] Running yarp name list ... found 3 ports
[MSG] ClusterWidget: yarprun successfully executed on icubsrv
[ERR] ClusterWidget: failed to run yarprun on icub-head

**Comment by jimingda** [comment]
I just found that the cpu light on the left back side of the icub can be pressed, after i press it the cpu color red become green.
I don't know it before   >.<

**Reply by jimingda** [reply]
After press the cpu button, things goes well, until i came across problem on following step, I cann't run the  "two iKinCartesian (for both arms) and iKinGazeCtrl."  and the Motor Indicator Light is still red, i wander when it will turn to green?

First, you will need to launch and prepare the robot. Go back to the tab 1/iCubStartUp.
    RUN only the two iKinCartesian (for both arms) and iKinGazeCtrl.

**Reply by jimingda** [reply, marked as answer]
I found that the Motor Indicator Light can be pressed  and then it will turn to green.
